pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManagerTest.java

```diff
@@ -939,6 +940,20 @@ public void testGetLiveBrokersForTable()
     for (String broker: liveBrokersForTable) {
       Assert.assertTrue(broker.startsWith("Broker_localhost"));
     }
+
+    // Test retrieving the live broker for table without table-type suffix.
+    liveBrokersForTable =
+        ControllerTestUtils.getHelixResourceManager().getLiveBrokersForTable(TABLE_NAME);
+    Assert.assertEquals(liveBrokersForTable.size(), 2);
+
+    // Test retrieving the live broker for table with non-existent table-type.
+    try {
+      ControllerTestUtils.getHelixResourceManager().getLiveBrokersForTable(REALTIME_TABLE_NAME);
+      Assert.fail("Method call above should have failed");
+    } catch (TableNotFoundException tableNotFoundException) {
+      Assert.assertTrue(tableNotFoundException.getMessage().contains(REALTIME_TABLE_NAME));
+    }
+
```

Comment:
Shall we add a test for cases where both OFFLINE and REALTIME table exist, or none of them exist?

Reply:
Done. I have duplicated the `waitForCondition` lambda definition with a change in the table-name. One other approach is to create a `Function` that takes in a string and returns another `Function` which can then be reused in the two waitForCondition calls. Lmk if we should remove the duplication.